source/AI.cpp

```diff
@@ -3365,7 +3370,32 @@ void AI::AimTurrets(const Ship &ship, FireCommand &command, bool opportunistic)
 				}
 
 				// Determine how much the turret must turn to face that vector.
-				double degrees = (Angle(p) - aim).Degrees();
+				double degrees = 0.0;
+				Angle angleToPoint = Angle(p);
+				if(hardpoint.IsOmnidirectional())
+					degrees = (angleToPoint - aim).Degrees();
+				else
+				{
+					Angle minArc = hardpoint.GetMinArc();
+					Angle maxArc = hardpoint.GetMaxArc();
+					const Angle facing = ship.Facing();
+					minArc += facing;
+					maxArc += facing;
+					if(!angleToPoint.IsInRange(minArc, maxArc))
+					{
+						// Decrease the priority of the target.
+						rendezvousTime += 2. * weapon->TotalLifetime();
+
+						// Point to the nearer edge of the arc.
+						const double degree1 = (minArc - angleToPoint).Degrees();
+						const double degree2 = (maxArc - angleToPoint).Degrees();
```

Comment:
numerals in a variable name? idk

Reply:
Fixed in [6b37ec5](https://github.com/endless-sky/endless-sky/pull/7094/commits/6b37ec5edf6e5b670e8dedb085d4ca58d0dcf9e3)